imap_processing/ialirt/l0/process_hit.py

```diff
@@ -0,0 +1,210 @@
+"""Functions to support HIT processing."""
+
+import logging
+from enum import Enum
+from typing import ClassVar
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+class HITPrefixes(Enum):
+    """Create ENUM for each rate type."""
+
+    FAST_RATE_1: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1A_TRIG", "IA_EVNT_TRIG", "A_EVNT_TRIG", "L3A_TRIG"]
+    ]
+    FAST_RATE_2: ClassVar[list[str]] = [
+        f"{prefix}_{i:02d}"
+        for i in range(15)
+        for prefix in ["L1B_TRIG", "IB_EVNT_TRIG", "B_EVNT_TRIG", "L3B_TRIG"]
+    ]
+    SLOW_RATE: ClassVar[list[str]] = [
+        "L1A",
+        "L2A",
+        "L3A",
+        "L1A0AHG",
+        "L1B0AHG",
+        "L1C0AHG",
+        "L4IAHG",
+        "L4OAHG",
+        "SLOW_RATE_08",
+        "SLOW_RATE_09",
+        "SLOW_RATE_10",
+        "L1A0BHG",
+        "L1B0BHG",
+        "L1C0BHG",
+        "L4IBHG",
+        "L4OBHG",
+        *[f"IALRT_RATE_{i}" for i in range(1, 21)],
+        "TRIG_IA_EVNT",
+        "TRIG_IB_EVNT",
+        "NASIDE_IALRT",
+        "NBSIDE_IALRT",
+        *[f"ERATE_{i}" for i in range(1, 6)],
+        "L12A",
+        "L123A",
+        "PENA",
+        "L12B",
+        "L123B",
+        "PENB",
+        "SLOW_RATE_51",
+        "SLOW_RATE_52",
+        "SLOW_RATE_53",
+        "SLOW_RATE_54",
+        "H_06_08",
+        "H_12_15",
+        "H_15_70",
+        "HE4_06_08",
+        "HE4_15_70",
+    ]
+
+
+def find_groups(data: xr.Dataset) -> xr.Dataset:
+    """
+    Find groups in the data.
```

Comment:
What does "groups" mean here?

Find all occurrences of the sequential set of 60 values 0 through 59. If a value is missing, or we are starting/ending in the middle of a sequence we do not count that as a valid group.